src/oneelement.jl

```diff
@@ -0,0 +1,27 @@
+"""
+    OneElement(val, ind, axes) <: AbstractArray
+Extremely simple `struct` used for the gradient of scalar `getindex`.
+"""
+struct OneElement{T,N,I,A} <: AbstractArray{T,N}
+  val::T
+  ind::I
+  axes::A
+  OneElement(val::T, ind::I, axes::A) where {T<:Number, I<:NTuple{N,Int}, A<:NTuple{N,AbstractUnitRange}} where {N} = new{T,N,I,A}(val, ind, axes)
+end
+
+OneElement(val, inds::NTuple{N,Int}, sz::NTuple{N,Integer}) where N = OneElement(val, inds, oneto.(sz))
+OneElement(val, inds::Int, sz::Int) = OneElement(val, (inds,), (sz,))
+OneElement(inds::Int, sz::Int) = OneElement(1, inds, sz)
+OneElement{T}(inds::Int, sz::Int) where T = OneElement(one(T), inds, sz)
+
+Base.size(A::OneElement) = map(length, A.axes)
+Base.axes(A::OneElement) = A.axes
+Base.getindex(A::OneElement{T,N}, i::Vararg{Int,N}) where {T,N} = ifelse(i==A.ind, A.val, zero(T))
```

Comment:
Should this have boundschecks? I didn't bother for Zygote, but perhaps for wider use... 

Fill does:
```julia
julia> Fill(1,2,3)
2×3 Fill{Int64}, with entries equal to 1

julia> ans[3,4]
ERROR: BoundsError: attempt to access 2×3 Fill{Int64} at index [3, 4]
```

Reply:
fixed!